app/services/db/identity/sp_active_user_counts_within_iaa_window.rb

```diff
@@ -0,0 +1,50 @@
+module Db
+  module Identity
+    # Similar to SpActiveUserCounts, but it limits dates to within active IAA windows
+    class SpActiveUserCountsWithinIaaWindow
+      def self.call
+        min_date, max_date = ServiceProvider.
+          where.not(iaa_start_date: nil).
+          where.not(iaa_end_date: nil).
+          pluck(:iaa_start_date, :iaa_end_date).
+          flatten.minmax
+
+        # issuer => ial => Set<UserIds>
+        by_issuer = Hash.new do |h, k|
+          h[k] = Hash.new { |h, k| h[k] = Set.new }
+        end
+
+        ::SpReturnLog.
+          where(returned_at: min_date..max_date).
+          includes(:service_provider).
+          find_in_batches(batch_size: 100_000) do |sp_return_logs|
```

Comment:
@stevegsa @mitchellhenke updated to do this query in Ruby, PTAL

Each iteration of this batch is <700ms to load 100K sp_return_logs and then about 1.4ms to eager-load the service_providers for that batch

The whole thing takes minutes, but there is a low chance of an individual query taking too long and getting timed out

Reply:
Yes I love one pass!

Reply:
Looks good to me!

Reply:
You want to use requested_at.  That field has an index.

Reply:
Using `requested_at` in 9beb18c379305f82f9a635d3785c274301b5d13a